### ISSUE TITLE
Vending machine will not give proper change when in Exact Change Only mode when customer inserts needed coins in same transaction

### ISSUE THREAD

**Issue by WoodyB** [WoodyB/vending-machine-project#532]
Version: 1.0.555

Scenario
Vending machine is in Exact Change Only state due to coin inventory of 10 quarters, 0 dimes and 0 nickels. The customer inserts 6 dimes and 1 nickel with the intention of purchasing candy for $0.65 but changes mind and decides to purchase chips for $0.50 instead.

Expected result
Vending machine keeps $0.50 for transaction, dispenses chips, returns $0.15 change and exits Exact Change Only state

Actual result
Vending machine keeps $0.65, dispenses chips and exits Exact Change Only state.

Steps to reproduce bug

1. Adjust start money in bin/app-data.js file 10 quarters, 0 dimes and 0 nickels
2. Start vending machine node bin/index
3. Observe, the vending machine will display Exact Change Only instead of the usual Insert Coin prompt.
4. Insert 6 dimes and 1 nickel
5. Select chips

**Comment by WoodyB** [comment]
Added a new acceptance test case for the scenario above. Once I trusted the test I looked into the fix. The issue was the design was coins in the current transaction were not added to the coins inventory until after the whole transaction was completed.

I changed this design so that after a coin has been accepted it is added to the coin inventory thus making it available to make change.

This bug fix will go out in next release. It appears something has broken in the CD pipeline and that will have to be fixed in order to get a new release out.
